### ISSUE TITLE
[debian] arbitrary `--user` throws permission errors

### ISSUE THREAD

**Issue by tobwen** [docker-library/postgres#917]
## tl;dr
I tried to run the example for arbitrary `--user` from the docs on the _Debian_-based container, but it threw several permission errors.

## error
```
chmod: changing permissions of '/var/lib/postgresql/data': Operation not permitted
chmod: changing permissions of '/var/run/postgresql': Operation not permitted
The files belonging to this database system will be owned by user "www-data".
This user must also own the server process.

The database cluster will be initialized with locale "en_US.utf8".
The default database encoding has accordingly been set to "UTF8".
The default text search configuration will be set to "english".

Data page checksums are disabled.

fixing permissions on existing directory /var/lib/postgresql/data ... initdb: error: could not change permissions of directory "/var/lib/postgresql/data": Operation not permitted
```

## steps to reproduce
See: https://github.com/docker-library/docs/blob/master/postgres/README.md#arbitrary---user-notes
``` shell
# running as root
$ docker run -it --rm --user www-data -e POSTGRES_PASSWORD=mysecretpassword postgres:bullseye

# other example
$ docker run -it --rm --user www-data:www-data -v /etc/passwd:/etc/passwd:ro -e POSTGRES_PASSWORD=mysecretpassword postgres:bullseye
$ grep www-data /etc/passwd
www-data:x:33:33:www-data:/var/www:/usr/sbin/nologin
```

## environment
_Docker version 20.10.12, build e91ed57_ from Docker's repository for Debian Buster (AMD64)

## analysis
Inside the container, there's already a `$PGDATA` directory, which belongs to user-id `999`. Of course, an arbitrary user cannot run `chmod` or other stuff here, as stated in https://github.com/docker-library/postgres/blob/ab940cbb923af99e2c7cf0e0ba5305bc6815aecc/14/bullseye/docker-entrypoint.sh#L41

``` shell
$ ls -n -a $PGDATA
total 8
drwxrwxrwx 2 999 999 4096 Dec 21 23:32 .
drwxr-xr-x 1 999 999 4096 Dec 21 23:32 ..

$ chmod 700 "$PGDATA" || :
chmod: changing permissions of '/var/lib/postgresql/data': Operation not permitted
```

## possible fix
Add `-e PGDATA=/tmp/pgdata` or another directory, where the user is allowed to write OR fix permissions of current location in the image.

**Comment by wglambert** [comment]
Yeah that error should be non-failing
https://github.com/docker-library/postgres/issues/853#issuecomment-853928725

**Comment by wglambert** [comment]
Closing since this seems resolved, if you need further help you could also try asking over at the [Docker Community Forums](https://forums.docker.com/), [Docker Community Slack](https://dockr.ly/slack), or [Stack Overflow](https://stackoverflow.com/search?tab=newest&q=docker)
